api/auth/azure/azure.go

```diff
@@ -172,23 +172,21 @@ func (a *AzureAuth) getJWT() (string, error) {
 	}
 	defer resp.Body.Close()
 
-	responseBytes, err := ioutil.ReadAll(resp.Body)
+	responseBytes, err := io.ReadAll(resp.Body)
 	if err != nil {
 		return "", fmt.Errorf("error reading response body from Azure token endpoint: %w", err)
 	}
 
 	if resp.StatusCode != http.StatusOK {
 		var errResp errorJSON
-		err = json.Unmarshal(responseBytes, &errResp)
-		if err != nil {
-			return "", fmt.Errorf("received error message but was unable to unmarshal its contents")
+		if err := json.Unmarshal(responseBytes, &errResp); err != nil {
+			return "", errors.New("received error message but was unable to unmarshal its contents")
```

Comment:
Although there are many places that we have returned `errors.New`, we would like to keep the `fmt.Errorf` for returning error from functions.

Reply:
Done